alacritty_terminal/src/term/mod.rs

```diff
@@ -398,13 +398,19 @@ impl<'a, C> Iterator for RenderableCellsIter<'a, C> {
                     let mut cell = RenderableCell::new(self, cell);
 
                     if self.cursor.key.style == CursorStyle::Block {
-                        // Invert cursor if static background is close to the cell's background.
-                        match self.cursor.cursor_color {
-                            CellRgb::Rgb(col) if col.contrast(cell.bg) >= MIN_CURSOR_CONTRAST => {
-                                cell.fg = self.cursor.text_color.color(cell.fg, cell.bg);
+                        cell.fg = match self.cursor.cursor_color {
+                            CellRgb::Rgb(col) => {
+                                // Invert cursor if static background is close to the cell's
+                                // background.
+                                if col.contrast(cell.bg) >= MIN_CURSOR_CONTRAST {
+                                    self.cursor.text_color.color(cell.fg, cell.bg)
+                                } else {
+                                    cell.bg
+                                }
                             },
-                            _ => cell.fg = cell.bg,
-                        }
+                            CellRgb::CellForeground => cell.fg,
+                            CellRgb::CellBackground => cell.bg,
+                        };
```

Comment:
Can you explain why you did not follow my suggestion here?

Reply:
I don't get your idea to translate to match statement. In what way do you want it to be? Like that?

```rust
match self.cursor.cursor_color {
  CellRgb::Rgb(col) => { /* something */ },
  _ => /* reset */,
}
```

Reply:
```
match self.cursor.cursor_color {
  CellRgb::Rgb(col) if col < MIN_CONTRAST => { /* do special thing */ },
  _ => /* do thing we've always done */,
}
```

Reply:
The only difference should be what happens when it's Rgb and the contrast is smaller than the minimum, all other cases just do the same thing as before.